ext/spl/spl_observer.c

```diff
@@ -942,62 +928,34 @@ SPL_METHOD(SplObjectStorage, __unserialize)
-	SPL_ME(SplObjectStorage,  setInfo,     arginfo_setInfo,       0)
-	SPL_ME(SplObjectStorage,  getHash,     arginfo_getHash,       0)
+	SPL_ME(SplObjectStorage,  attach,      arginfo_class_SplObjectStorage_attach,        0)
+	SPL_ME(SplObjectStorage,  detach,      arginfo_class_SplObjectStorage_detach,        0)
+	SPL_ME(SplObjectStorage,  contains,    arginfo_class_SplObjectStorage_contains,        0)
+	SPL_ME(SplObjectStorage,  addAll,      arginfo_class_SplObjectStorage_addAll,        0)
+	SPL_ME(SplObjectStorage,  removeAll,   arginfo_class_SplObjectStorage_removeAll,        0)
+	SPL_ME(SplObjectStorage,  removeAllExcept,   arginfo_class_SplObjectStorage_removeAllExcept,  0)
+	SPL_ME(SplObjectStorage,  getInfo,     arginfo_class_SplObjectStorage_getInfo,0)
+	SPL_ME(SplObjectStorage,  setInfo,     arginfo_class_SplObjectStorage_setInfo,       0)
+	SPL_ME(SplObjectStorage,  getHash,     arginfo_class_SplObjectStorage_getHash,       0)
 	/* Countable */
-	SPL_ME(SplObjectStorage,  count,       arginfo_splobject_void,0)
+	SPL_ME(SplObjectStorage,  count,       arginfo_class_SplObjectStorage_count,0)
 	/* Iterator */
-	SPL_ME(SplObjectStorage,  rewind,      arginfo_splobject_void,0)
-	SPL_ME(SplObjectStorage,  valid,       arginfo_splobject_void,0)
-	SPL_ME(SplObjectStorage,  key,         arginfo_splobject_void,0)
-	SPL_ME(SplObjectStorage,  current,     arginfo_splobject_void,0)
-	SPL_ME(SplObjectStorage,  next,        arginfo_splobject_void,0)
+	SPL_ME(SplObjectStorage,  rewind,      arginfo_class_SplObjectStorage_rewind,0)
+	SPL_ME(SplObjectStorage,  valid,       arginfo_class_SplObjectStorage_valid,0)
+	SPL_ME(SplObjectStorage,  key,         arginfo_class_SplObjectStorage_key,0)
+	SPL_ME(SplObjectStorage,  current,     arginfo_class_SplObjectStorage_current,0)
+	SPL_ME(SplObjectStorage,  next,        arginfo_class_SplObjectStorage_next,0)
 	/* Serializable */
-	SPL_ME(SplObjectStorage,  unserialize, arginfo_Serialized,    0)
-	SPL_ME(SplObjectStorage,  serialize,   arginfo_splobject_void,0)
-	SPL_ME(SplObjectStorage,  __unserialize, arginfo_Serialized,    0)
-	SPL_ME(SplObjectStorage,  __serialize,   arginfo_splobject_void,0)
+	SPL_ME(SplObjectStorage,  unserialize, arginfo_class_SplObjectStorage_unserialize,    0)
+	SPL_ME(SplObjectStorage,  serialize,   arginfo_class_SplObjectStorage_serialize,0)
+	SPL_ME(SplObjectStorage,  __unserialize, arginfo_class_SplObjectStorage___unserialize,    0)
+	SPL_ME(SplObjectStorage,  __serialize,   arginfo_class_SplObjectStorage___serialize,0)
 	/* ArrayAccess */
-	SPL_MA(SplObjectStorage, offsetExists, SplObjectStorage, contains, arginfo_offsetGet, 0)
-	SPL_MA(SplObjectStorage, offsetSet,    SplObjectStorage, attach,   arginfo_attach, 0)
-	SPL_MA(SplObjectStorage, offsetUnset,  SplObjectStorage, detach,   arginfo_offsetGet, 0)
-	SPL_ME(SplObjectStorage, offsetGet,    arginfo_offsetGet,     0)
+	SPL_MA(SplObjectStorage, offsetExists, SplObjectStorage, contains, arginfo_class_SplObjectStorage_offsetExists, 0)
```

Comment:
I think there is a problem with these `offset*()` methods: they are aliased to methods whose ZPP is not compatible with `ArrayOffset::offset*()` methods. Or it isn't a problem?

Reply:
I think it's fine. Think of it as `function offsetGet($x) { if (!is_object($x)) { throw new Exception; } } }`. Especially for interfaces like ArrayAccess, which would need generics to be well-defined, having a non-signature restriction on the acceptable values is common.